src/federationparams.cpp

```diff
@@ -139,21 +140,24 @@ CFederationParams::CFederationParams(const int networkId, const std::string data
     vFixedSeeds = std::vector<SeedSpec6>(pnSeed6_main, pnSeed6_main + ARRAYLEN(pnSeed6_main));
 }
 
-CPubKey CFederationParams::ReadAggregatePubkey(const std::vector<unsigned char>& pubkey)
+CPubKey CFederationParams::ReadAggregatePubkey(const std::vector<unsigned char>& pubkey, uint height)
 {
     if(!pubkey.size())
         throw std::runtime_error("Aggregate Public Key for Signed Block is empty");
     
     if (pubkey[0] == 0x02 || pubkey[0] == 0x03) {
-        aggregatePubkey = CPubKey(pubkey.begin(), pubkey.end());
-        if(!aggregatePubkey.IsFullyValid()) {
+        aggPubkeyAndHeight p;
+        p.aggpubkey = CPubKey(pubkey.begin(), pubkey.end());
+        p.height = height;
+        aggregatePubkeyHeight.push_back(p);
```

Comment:
We have to handle this call cautiously. 
I think we need to move this line:
`aggregatePubkeyHeight.push_back(p); `
do it only after the p.aggpubkey is validated. Otherwise the list is going to have an invalid aggpubkey.

If `p.aggpubkey.IsFullyValid() ` fails while loading genesis block its ok to throw std::runtime_error which stops the node. we cannot start the node without correct genesis block. When we are trying to setup the network this is ok.

But when the network is online and a new federation block is being processed, if it has invalid aggpubkey we should not throw. we have to reject the block and continue. otherwise the node will stop and could bring the whole network down!

Reply:
i will move it after
`if(!p.aggpubkey.IsFullyValid())` and
`if (p.aggpubkey.size() != CPubKey::COMPRESSED_PUBLIC_KEY_SIZE)`
before `return`